src/service.ts

```diff
@@ -47,6 +54,30 @@ export class PisaService {
             next();
         });
 
+        // rate limits
+        if (config && config.rateGlobal) {
+            app.use(new rateLimit({
+                keyGenerator: () => "global", // use the same key for all users
+                statusCode: 503, // = Too Many Requests (RFC 7231)
+                message: "PISA is overloaded right now. Please try again later.",
+                ...config.rateGlobal
+            }));
+            logger.info(`PISA api global rate limit: ${config.rateGlobal.max} requests every: ${config.rateGlobal.windowMs/1000} seconds.`);
+        } else {
+            logger.warn(`PISA api global rate limit: NOT SET.`);
+        }
+
+        if (config && config.ratePerUser) {
+            app.use(new rateLimit({
+                statusCode: 429, // = Too Many Requests (RFC 6585)
```

Comment:
I know the default is already to IP, but it would be easier to read if we put the keyGenerator value in: keyGenerator: (req) => req.ip.

Reply:
Done